src/model/connectionNode.ts

```diff
@@ -70,4 +79,37 @@ export class ConnectionNode implements INode {
 
         mysqlTreeDataProvider.refresh();
     }
+
+    public async editConnection(context: vscode.ExtensionContext, mysqlTreeDataProvider: MySQLTreeDataProvider) {
+        AppInsightsClient.sendEvent("editConncetion.start");
+
+        const copyFrom: IConnection = {
+            id: this.id,
+            host: this.host,
+            user: this.user,
+            password: this.password,
+            port: this.port,
+            certPath: this.certPath,
+        };
+        const editConnection = await Utility.createConnectionFromInput(context, copyFrom);
+
+        if (editConnection !== undefined) {

```

Comment:
This logic is similar to addConnection. Please help merge them.

Reply:
Merge them into a addOrEditConnection method?
Should the metod(s) be placed on the connectionNode or the mysqlTreeDataProvider class?

Reply:
We could wrap these similar code snippet into a method into https://github.com/formulahendry/vscode-mysql/blob/master/src/common/utility.ts

Reply:
(We still keep editConnection in connectionNode and addConnection in mysqlTreeDataProvider, and call shared method in utility)